### ISSUE TITLE
Bug: Error parsing typescript config file

### ISSUE THREAD

**Issue by deathgaze** [eslint/eslint#16279]
### Environment

Node version: v16.13.1
npm version: v8.1.2
Local ESLint version: v8.4.1 (Currently used)
Global ESLint version: v8.23.0
Operating System: darwin 21.2.0

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
{
  "root": true,
  "extends": "../../.eslintrc.json",
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts"],
      "parserOptions": {
        "project": [
          "projects/app/tsconfig.app.json",
          "projects/app/tsconfig.spec.json"
        ],
        "createDefaultProgram": true
      },
      "rules": {
        "@angular-eslint/directive-selector": [
          "error",
          {
            "type": "attribute",
            "prefix": "app",
            "style": "camelCase"
          }
        ],
        "@angular-eslint/component-selector": [
          "error",
          {
            "type": "element",
            "prefix": "app",
            "style": "kebab-case"
          }
        ]
      }
    },
    {
      "files": ["*.html"],
      "rules": {}
    }
  ]
}

```

</details>

```js
<!-- Paste your code here -->
```

### What did you expect to happen?

ESLint should parse the TS configuration file in the same manner that Angular's `lint` script does.

### What actually happened?

All files error out with an error similar to this:
```
/Users/208115/dev/butterfly-ui/projects/app/src/environments/environment.ts
  0:0  error  Parsing error: Cannot read file '/users/208115/dev/butterfly-ui/projects/app/projects/app/tsconfig.app.json'
```

output of `eslint . --debug` follows
```
eslintrc:cascading-config-array-factory Load config files for /Users/208115/dev/butterfly-ui/projects/app/src. +2ms
  eslintrc:cascading-config-array-factory Cache hit: /Users/208115/dev/butterfly-ui/projects/app/src. +0ms
  eslint:cli-engine Lint /Users/208115/dev/butterfly-ui/projects/app/src/test.ts +3ms
  eslint:linter Linting code for /Users/208115/dev/butterfly-ui/projects/app/src/test.ts (pass 1) +1ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/208115/dev/butterfly-ui/projects/app/src/test.ts +0ms
  eslint:linter Apply the processor: '@angular-eslint/template/extract-inline-html' +0ms
  eslint:linter A code block was found: '(unnamed)' +0ms
  eslint:linter Parsing: /Users/208115/dev/butterfly-ui/projects/app/src/test.ts +1ms
  eslint:linter Parsing error: Cannot read file '/users/208115/dev/butterfly-ui/projects/app/projects/app/tsconfig.app.json'.
  eslint:linter Error: Cannot read file '/users/208115/dev/butterfly-ui/projects/app/projects/app/tsconfig.app.json'.
    at Object.diagnosticReporter [as onUnRecoverableConfigFileDiagnostic] (/Users/208115/dev/butterfly-ui/node_modules/@typescript-eslint/parser/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createWatchProgram.js:95:11)
    at Object.getParsedCommandLineOfConfigFile (/Users/208115/dev/butterfly-ui/node_modules/typescript/lib/typescript.js:39854:18)
    at parseConfigFile (/Users/208115/dev/butterfly-ui/node_modules/typescript/lib/typescript.js:120763:43)
    at Object.createWatchProgram (/Users/208115/dev/butterfly-ui/node_modules/typescript/lib/typescript.js:120371:13)
    at createWatchProgram (/Users/208115/dev/butterfly-ui/node_modules/@typescript-eslint/parser/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createWatchProgram.js:286:22)
    at getProgramsForProjects (/Users/208115/dev/butterfly-ui/node_modules/@typescript-eslint/parser/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createWatchProgram.js:184:30)
    at createProjectProgram (/Users/208115/dev/butterfly-ui/node_modules/@typescript-eslint/parser/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.js:22:106)
    at getProgramAndAST (/Users/208115/dev/butterfly-ui/node_modules/@typescript-eslint/parser/node_modules/@typescript-eslint/typescript-estree/dist/parser.js:87:61)
    at parseAndGenerateServices (/Users/208115/dev/butterfly-ui/node_modules/@typescript-eslint/parser/node_modules/@typescript-eslint/typescript-estree/dist/parser.js:488:31)
    at Object.parseForESLint (/Users/208115/dev/butterfly-ui/node_modules/@typescript-eslint/parser/dist/parser.js:97:80) +2ms
```

### Participation

- [ ] I am willing to submit a pull request for this issue.
